python-packages/smithy-core/smithy_core/documents.py

```diff
@@ -284,3 +310,137 @@ def __eq__(self, other: object) -> bool:
         # JsonDocument, cares about the schema then it's responsible for transforming
         # the output of `as_value` to the canonical form.
         return self.as_value() == other.as_value()
+
+
+class DocumentSerializer(ShapeSerializer):
```

Comment:
Is there any way to prevent access to this directly and hide it all behind a method that internally uses this? That way you can completely change how it works in the future and even remove this code if you really wanted to. Or I guess you just add a "_" to the name to say use at your own risk.

Reply:
No way to prevent access I'm afraid, but I did prefix it